lib/_stream_writable.js

```diff
@@ -53,6 +53,46 @@ Object.setPrototypeOf(Writable, Stream);
 
 function nop() {}
 
+const REQUEST_POOL = [];
```

Comment:
How much throughput is this going to give us? I'm a bit worried because objects in this array are going to end up in old space and they will be referring to objects in new space. I've used this technique in the past and it's quite powerful, however I'm worried this _might_ look like a memory leak to people looking at the code.

Have you measured if this is giving us any perf benefits at all?

Reply:
I don't think our benchmarks are good enough to show either way... object allocations in V8 are very cheap... what we win here mostly is avoiding GC pauses... not sure how to benchmark that...

Reply:
I think Matteo's concern is more about the increased memory usage that this pool would create when there are a lot of streams (think of persistent connections) rather than allocations.